tests/system/python/packages/test_pi_webapi_linked_data_type.py

```diff
@@ -0,0 +1,277 @@
+    for plugin in SOUTH_PLUGINS_LIST:
+        assert "{}".format(plugin) in list_of_filters
+
+def verify_service_added(fledge_url, ):
+    get_url = "/fledge/south"
+    result = utils.get_request(fledge_url, urllib.parse.quote(get_url, safe='?,=,&,/'))['services']
+    assert len(result)
+    list_of_southbounds = list(map(lambda val: val['name'], result))
+    for plugin in SOUTH_PLUGINS_LIST:
+        assert "{}_{}".format(ASSET, plugin) in list_of_southbounds
+
+    get_url = "/fledge/north"
+    result = utils.get_request(fledge_url, get_url)
+    assert len(result)
+    list_of_northbounds = list(map(lambda val: val['name'], result))
+    assert NORTH_INSTANCE_NAME in list_of_northbounds
+
+    get_url = "/fledge/service"
+    result = utils.get_request(fledge_url, urllib.parse.quote(get_url, safe='?,=,&,/'))['services']
+    assert len(result)
+    list_of_services = list(map(lambda val: val['name'], result))
+    for plugin in SOUTH_PLUGINS_LIST:
+        assert "{}_{}".format(ASSET, plugin) in list_of_services
+    assert NORTH_INSTANCE_NAME in list_of_services
+
+def verify_data_between_fledge_and_piwebapi(fledge_url, pi_host, pi_admin, pi_passwd, pi_db, AF_HIERARCHY_LEVEL, ASSET, SOUTH_PLUGINS_LIST, verify_hierarchy_and_get_datapoints_from_pi_web_api):
+    # Checking if hierarchy created properly or not, and retrieveing data from PI Server
+    print("Values --> ", fledge_url, pi_host, pi_admin, pi_passwd, pi_db, AF_HIERARCHY_LEVEL, ASSET, SOUTH_PLUGINS_LIST)
+    data_from_pi = verify_hierarchy_and_get_datapoints_from_pi_web_api(pi_host, pi_admin, pi_passwd, pi_db, AF_HIERARCHY_LEVEL, ASSET, SOUTH_PLUGINS_LIST)
+    print(data_from_pi)
+    assert len(data_from_pi) >= len(SOUTH_PLUGINS_LIST), "Datapoint does not exist"
+    print('Data from PI Web API')
+    print(data_from_pi)
+    
+    data_from_fledge = get_data_from_fledge(fledge_url)
+    print('data fledge retrieved')
+    print(data_from_fledge)
+    
+    # For verifying data send to PI Server from fledge is same.
+    check_data = verify_equality_between_fledge_and_pi(data_from_fledge, data_from_pi)
+    assert check_data == 'Equal', "Data is not equal"
+
+@pytest.fixture
+def add_configure_filter(add_filter, fledge_url):
+    for plugin in SOUTH_PLUGINS_LIST:
+        filter_cfg = {"enable": "true", "expression": "log({}_{}-Ingest)".format(ASSET, plugin)}
+        add_filter("expression", None, "{}".format(plugin), filter_cfg, fledge_url, "{}_{}".format(ASSET, plugin), installation_type='package')
+
+@pytest.fixture
+def start_south_north(add_south, start_north_task_omf_web_api, add_filter, remove_data_file,
+                      fledge_url, pi_host, pi_port, pi_admin, pi_passwd, pi_db,
+                      start_north_omf_as_a_service, enable_schedule, 
+                      clear_pi_system_through_pi_web_api, asset_name=ASSET):
+    """ This fixture starts the sinusoid plugin and north pi web api plugin. Also puts a filter
```

Comment:
Are we starting only sinusoid plugin?

Reply:
No, we are using siunsoid and randomwalk and starting both the plugins.